include/geopackage/SQLite.hpp

```diff
@@ -0,0 +1,273 @@
+#ifndef NGEN_GEOPACKAGE_SQLITE_H
+#define NGEN_GEOPACKAGE_SQLITE_H
+
+#include <memory>
+#include <stdexcept>
+#include <vector>
+#include <string>
+
+#include <sqlite3.h>
+
+namespace geopackage {
+
+/**
+ * Runtime error for iterations that haven't started
+ */
+const auto sqlite_get_notstarted_error = std::runtime_error(
+    "sqlite iteration is has not started, get() is not callable (call sqlite_iter::next() before)"
+);
+
+/**
+ * Runtime error for iterations that are finished
+ */
+const auto sqlite_get_done_error = std::runtime_error(
+    "sqlite iteration is done, get() is not callable"
+);
+
+/**
+ * Deleter used to provide smart pointer support for sqlite3 structs.
+ */
+struct sqlite_deleter
+{
+    void operator()(sqlite3* db) { sqlite3_close_v2(db); }
+    void operator()(sqlite3_stmt* stmt) { sqlite3_finalize(stmt); }
+};
+
+/**
+ * Smart pointer (unique) type for sqlite3 database
+ */
+using sqlite_t = std::unique_ptr<sqlite3, sqlite_deleter>;
```

Comment:
This type alias has pretty limited use, and only in code private to `class sqlite` itself. Let's move it inside that, or drop it entirely.

Reply:
Added suggestion in 49bd6d9.